cli-client/helpers/tts_client.py

```diff
@@ -24,19 +36,109 @@ def _compose_synthesis_request(self, text: str, voice: str, audio_format: str, s
         )
 
         return message
+    
+    def _save_audio_result(self, audio_samples: bytes):
+        audio_exporter = AudioExporter(self._audio_sample_rate)
+        audio_exporter.save_audio(self._audio_format, audio_samples, self._audio_file)
+        logging.info("Stored synthesis audio at -%s-", self._audio_file)
 
-    def synthesize(self, text: str, voice: str) -> bytes:
-        logging.info("Sending synthesis request for voice: -%s-, sampling_rate: %i and text: -%s-", voice, self._audio_sample_rate, text)
+    def synthesize(self):
+        logging.info("Sending synthesis request for voice: -%s-, sampling_rate: %i and text: -%s-", self._voice, self._audio_sample_rate, self._text)
+        selected_audio_format = AudioExporter.SUPPORTED_FORMATS[self._audio_format]
 
         metadata = None if self._secure_channel else [('authorization', "Bearer " + self._token)]
         response, call = self._stub.SynthesizeSpeech.with_call(
             self._compose_synthesis_request(
-                text=text,
-                voice=voice,
+                text=self._text,
+                voice=self._voice,
                 sampling_rate=self._audio_sample_rate,
-                audio_format=self._audio_format
+                audio_format=selected_audio_format
             ), metadata=metadata
         )
 
         logging.info("Synthesis response [status=%s]", str(call.code()))
-        return response.audio_samples
+        logging.info("Received response with %s bytes of audio data", len(response.audio_samples))
+        self._save_audio_result(response.audio_samples)
+
+    def _close_stream_by_inactivity(self):
+        logging.info("Stream inactivity detected, closing stream...")
+        self._peer_responded.set()
+
+    def _start_inactivity_timer(self, inactivity_timeout: float):
+        self._inactivity_timer = Timer(inactivity_timeout, self._close_stream_by_inactivity)
+        self._inactivity_timer.start()
+    
+    def _response_watcher(self, response_iterator):
+        try:
+            audio = bytearray()
+            logging.info("Running response watcher")
+            for response in response_iterator:
+                logging.debug("New incoming response of type: %s", type(response))
+                
+                if response.streaming_audio.audio_samples:
+                    logging.info("StreamingAudio response received: %s bytes of audio data", len(response.streaming_audio.audio_samples))
+                    audio.extend(response.streaming_audio.audio_samples)
+                
+                if response.end_of_utterance.data:
+                    logging.info("EndOfUtterance response received. Signaling end of stream with data: %s", response.end_of_utterance.data)
+
+                if self._inactivity_timer:
+                    self._inactivity_timer.cancel()
+                self._start_inactivity_timer(self._inactivity_timer_timeout)
+                
+            self._save_audio_result(audio)
```

Comment:
We should probably move this method `_save_audio_result` out of the TTS Client (probably into `cli-client/synthesizer_stream.py`) and just return the audio vector as a result. That way, it would be easier to test and we wouldn't have to include any patches in the unit tests.